### PR TITLE
Staramr/search

### LINKED ISSUE #3755: new module: staramr/search
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

*Note: The issue #1160 is for adding staramr, but it was closed due to inactivity so I am opening a new issue.*

### Are you going to work on this?

- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by apetkau**
This adds a new module for the [staramr](https://github.com/phac-nml/staramr) tool.

## PR checklist

Closes #3755 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- ~[ ] If necessary, include test data in your PR.~
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  -  [x] \[I added this\] tests are provided by nf-test (running `nf-core modules test staramr/search` works for docker, singularity, and conda)
  - ~[ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`~
  - ~[ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`~
  - ~[ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`~

### CODE REVIEW

File: modules/nf-core/staramr/search/main.nf [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    output:
+    tuple val(meta), path("*_results/results.xlsx")           , emit: results_xlsx
+    tuple val(meta), path("*_results/summary.tsv.gz")         , emit: summary_tsv
+    tuple val(meta), path("*_results/detailed_summary.tsv.gz"), emit: detailed_summary_tsv
+    tuple val(meta), path("*_results/resfinder.tsv.gz")       , emit: resfinder_tsv
+    tuple val(meta), path("*_results/plasmidfinder.tsv.gz")   , emit: plasmidfinder_tsv
+    tuple val(meta), path("*_results/mlst.tsv.gz")            , emit: mlst_tsv
+    tuple val(meta), path("*_results/settings.txt.gz")        , emit: settings_txt
+    tuple val(meta), path("*_results/pointfinder.tsv.gz")     , emit: pointfinder_tsv, optional: true
```
Comment (erinyoung) [inline modules/nf-core/staramr/search/main.nf:21]: Do these files need to be compressed? 

I thought it was was only large files (i.e. sam, fasta, fastq) that needed to be compressed.

Comment (apetkau) [reply modules/nf-core/staramr/search/main.nf:21]: Technically, not compressed. I wasn't actually sure if all outputs needed to be compressed. If it's only large files I can remove the compression code.

Comment (ggabernet) [reply modules/nf-core/staramr/search/main.nf:21]: ah yes right, I misunderstood earlier when you asked about that. Compression would be needed only for bigger files e.g. fastq

Comment (apetkau) [reply modules/nf-core/staramr/search/main.nf:21]: Ah, okay. Thanks so much 😄 . Removed gzipping small output files in 33f0f6e68de39a35c9a3d7dde30cc2872f100663

**Review by ggabernet — APPROVED** [review]
Looks good to me now!

File: modules/nf-core/staramr/search/main.nf (hunk 1 above)
Comment (apetkau) [reply modules/nf-core/staramr/search/main.nf]: Woops. This commit is also needed for removing gzip: 33f0f6e68de39a35c9a3d7dde30cc2872f100663

I made those changes in my editor to `main.nf` and forgot to save the file and commited everything else 😅

**Comment by ggabernet** [comment]
Hi @adamrtalbot , could we have a review here for this module contributed during the Boston hackathon? 🙏

**Review by erinyoung — APPROVED** [review]
Looks good to me

**Pull request merged**